illumina_somaticVariants.pm

```diff
@@ -30,12 +31,23 @@ sub parseSamples {
             }
         }
 
-        if ($origin =~ m/R.*/){
-            push(@{$somatic_samples{$cpct_name}{"ref"}}, $sample);
-        }
+        if($opt{SET_HAS_METADATA} eq "yes"){
+            print "Reading Tumour/Reference names from metadata\n\n";
+            my $metadata = metadataParse($opt{OUTPUT_DIR});
+            # Reference sample
+            push(@{$somatic_samples{$cpct_name}{"ref"}}, $metadata->{'ref_sample'});
+            # Tumour sample
+            push(@{$somatic_samples{$cpct_name}{"tumor"}}, $metadata->{'tumor_sample'});
+         } else {
+                # Reference sample
+                if ($origin =~ m/R.*/){
+                    push(@{$somatic_samples{$cpct_name}{"ref"}},$sample);
+                }
 
-        elsif ($origin =~ m/T.*/){
-            push(@{$somatic_samples{$cpct_name}{"tumor"}}, $sample);
+                # Tumor samples
+                elsif ($origin =~ m/T.*/){
```

Comment:
Project style is already a bit inconsistent, but take the opportunity to move this up to be `} elsif () {`